docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,104 @@ Note that:
\ No newline at end of file
+  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` supports two aspects:
+1) Allowing to associate a configuration file with a cloud's child-device twin
+2) Allowing to consume/provide a configuration file from/to an external device via network
+
+**Details to Aspect 1: Associating to cloud's child-device twin**
+
+For aspect (1) the plugin provides the field `childid` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). That field is interpreted as unique child-device id and the plugin associates the record's configuration file with corresponding cloud's child-device twin. If not provided the configuration file is associated with cloud's thin-edge device twin.
+
+Example:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },              # appears in the cloud on the thin-edge device
+    { path = '/etc/child1/foo.conf', type = 'foo', childid = 'child1'  }  # appears in the cloud on the child-device 'child1'
+  ]
+```
+
+Note that the `c8y_configuration_plugin` does not create any child-device twin in the cloud. Instead the clouds child-device twin must be created upfront. However the plugin takes care to set all capabilities (supported operations, config-types) on that child-device twin in the cloud in the same way as it does for the thin-edge device.
+
+
+**Details to Aspect 2: Consuming/providing configuration files via network**
+
+For aspect (2) there are two proposals as below. Decision has to been taken which proposal to follow.
+
+**Proposal 1:**
+
+The `c8y_configuration_plugin` configuration's record field `path` (reference to section 'Configuration' above) can be prefixed with "mqtt://". Then it's value is treated as MQTT topic structure, where another process/external device is expected to provide/consume the configuration file.
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file as retain message to the MQTT topic `{path}`. Then the plugin consumes that retain message whenever a cloud request comes in.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+    { path = 'mqtt://configs/bar.conf', type = 'bar.conf', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: publishs all its config files to MQTT (with retain); Example: Publish to topic `configs/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
```

Comment:
This document behind a reference, I would expect the format of this request.

Reply:
That is the SmartREST request coming from C8Y, that instructs the plugin to upload a config file. So the format of that request is given by the C8Y API, as I can see SmartREST messages 526.

Reply:
Yes, but it would be good to have here the description of this 526 SmartRest message. This is a reference doc and users/testers/devs should not have to figure out themselves the details searching the documentation of cumulocity.

Reply:
Added description of C8Y SmartREST messages. See https://github.com/cstoidner/thin-edge.io/commit/2f5deb463a1d5ccafc86c458c3815ae90024d9ff